includes/Admin/Product_Editor_Compatibility.php

```diff
@@ -0,0 +1,181 @@
+		$response->set_data( $data );
+
+		return $response;
+	}
+
+	/**
+	 * Saves custom data to product metadata.
+	 *
+	 * @param WC_Product      $product The product object.
+	 * @param WP_REST_Request $request The request object.
+	 */
+	public function process_data_before_save( $product, $request ) {
+		$is_square_synced = $request->get_param( 'is_square_synced' );
+		$stock_quantity   = $request->get_param( 'stock_quantity' );
+
+		if ( ! is_null( $is_square_synced ) ) {
+			if ( $is_square_synced ) {
+				Product::set_synced_with_square( $product, 'yes' );
+			} else {
+				Product::set_synced_with_square( $product, 'no' );
+			}
+		}
+
+		if ( ! is_null( $stock_quantity ) && ! empty( $stock_quantity ) ) {
+			$product->set_stock_quantity( $stock_quantity );
+		}
+	}
+
+	/**
+	 * Adds the sync with Square control to the product editor.
+	 *
+	 */
+	public function add_sync_with_square_control( $basic_details_field ) {
+		$basic_details_field->add_block(
+			array(
+				'id'             => '_wc_square_synced',
+				'blockName'      => 'woocommerce-square/sync-with-square-field',
+				'attributes'     => array(
+					'title' => __( 'Sync with Square', 'woocommerce-square' ),
+				),
+				'hideConditions' => array(
+					array(
+						'expression' => '!editedProduct.is_sync_enabled',
+					),
+				),
+			)
+		);
+	}
+
+	/**
+	 * Adds the inventory control to the product editor.
+	 *
+	 * @param Section $sku_field The SKU field block.
+	 */
+	public function add_inventory_control( $sku_field ) {
+		/**
+		 * Inventory section block.
+		 *
+		 * @var Section $parent
+		 */
+		$parent = $sku_field->get_parent();
+
+		$parent->add_block(
+			array(
+				'id'         => '_wc_square_stock_management_field',
+				'blockName'  => 'woocommerce-square/stock-management-field',
+				'attributes' => array(
+					'disabled'     => 'yes' !== get_option( 'woocommerce_manage_stock' ),
+					'disabledCopy' => sprintf(
+						/* translators: %1$s: Learn more link opening tag. %2$s: Learn more link closing tag.*/
+						__( 'Per your %1$sstore settings%2$s, inventory management is <strong>disabled</strong>.', 'woocommerce-square' ),
+						'<a href="' . admin_url( 'admin.php?page=wc-settings&tab=products&section=inventory' ) . '" target="_blank" rel="noreferrer">',
+						'</a>'
+					),
+				),
+			)
+		);
+
+		$parent->add_block(
+			array(
+				'id'        => '_wc_square_stock_quantity_field',
+				'blockName' => 'woocommerce-square/stock-quantity-field',
+			)
+		);
+	}
+
+	/**
+	 * Removes the manage stock and stock quantity blocks from the product editor.
+	 *
+	 * @param ProductBlock $block Core product blocks.
+	 */
+	public function remove_core_blocks( $block ) {
+		$blocks_to_remove = array(
+			'product-inventory-quantity',
+			'product-track-stock',
+		);
+
+		if ( in_array( $block->get_id(), $blocks_to_remove, true ) ) {
+			$block->remove(); // This is a workaround and is not recommended by Woo.
```

Comment:
Do we have more information on why we're taking this approach if it's not recommended?

Reply:
@dkotter in the classic editor, the plugin modifies the functionality of some of the core product meta fields. In the new product editor, we don't have a way to modify them however we can remove them and implement our own custom implementation.

For example, in the classic editor we conditionally disable the inventory tracking field which at the moment we cannot do in the new editor.

Reply:
I think it would be worth updating the comment here to note this, rather than just saying this isn't recommended, just so there's no confusion by anyone else that views this